alibi/explainers/tests/test_partial_dependence.py

```diff
@@ -0,0 +1,579 @@
+
+    if n_ice == 'all':
+        ice_sampled_vals = _sample_ice(ice_values=ice_vals, n_ice=n_ice, seed=0)
+        np.testing.assert_allclose(ice_vals, ice_sampled_vals)
+
+    elif n_ice == 'list':
+        size = np.random.randint(1, n_samples)
+        # needs to be sorted because of the np.unique applied inside the _sample_ice
+        n_ice = np.sort(np.random.choice(n_samples, size=size, replace=False)).tolist()
+        ice_sampled_vals = _sample_ice(ice_values=ice_vals, n_ice=n_ice, seed=0)
+        np.testing.assert_allclose(ice_vals[:, n_ice], ice_sampled_vals)
+
+    else:
+        n_ice = np.random.randint(1, n_samples)
+        ice_sampled_vals = _sample_ice(ice_values=ice_vals, n_ice=n_ice, seed=0)
+        assert ice_sampled_vals.shape[1] == n_ice
+
+
+@pytest.mark.parametrize('n_ice', ['unknown', -10, [-1, 1, 2]])
+@pytest.mark.parametrize('n_samples, n_values', [(100, 10)])
+def test_ice_sampling_error(n_samples, n_values, n_ice):
+    """ Checks if the ice sampling helper function throws an error when the arguments are invalid. """
+    ice_vals = np.random.rand(n_values, n_samples)
+    with pytest.raises(ValueError):
+        _sample_ice(ice_values=ice_vals, n_ice=n_ice, seed=0)
+
+
+@pytest.mark.parametrize('rf_classifier', [lazy_fixture('iris_data')], indirect=True)
+@pytest.mark.parametrize('features', [
+    [0], [1], [2],
+    [(0, 1)], [(0, 2)]
+])
+@pytest.mark.parametrize('params', [
+    {
+        'response_method': 'auto',
+        'percentiles': (0, 1),
+        'grid_resolution': 10,
+        'method': 'auto',
+        'kind': 'average'
+    }
+])
+def test_sklearn_numerical(rf_classifier, iris_data, features, params):
+    """ Checks alibi pd implementation against the sklearn implementation for numerical features."""
+    rf, _ = rf_classifier
+    X_train = iris_data['X_train']
+
+    # compute pd with alibi
+    explainer = PartialDependence(predictor=rf)
+    exp_alibi = explainer.explain(X=X_train, features_list=features, **params)
+
+    # compute pd with sklearn
+    exp_sklearn = partial_dependence(X=X_train, estimator=rf, features=features, **params)
+
+    assert np.allclose(exp_alibi.pd_values[0], exp_sklearn['average'])
+    if isinstance(features[0], numbers.Integral):
+        assert np.allclose(exp_alibi.feature_values[0], exp_sklearn['values'][0])
+    else:
+        for i in range(len(exp_sklearn['values'])):
+            assert np.allclose(exp_alibi.feature_values[0][i], exp_sklearn['values'][i])
+
+
+@pytest.mark.parametrize('rf_classifier', [lazy_fixture('adult_data')], indirect=True)
+@pytest.mark.parametrize('features', [
+    [1], [2], [4], [5],
+    [(1, 2)], [(2, 4)], [(4, 5)]
+])
+@pytest.mark.parametrize('params', [
+    {
+        'response_method': 'auto',
+        'percentiles': (0, 1),
+        'grid_resolution': np.inf,
+        'method': 'auto',
+        'kind': 'average'
+    }
+])
+def test_sklearn_categorical(rf_classifier, adult_data, features, params):
```

Comment:
So `sklearn` can compute categorical PDs correctly as long as `grid_resolution` is high enough?

Reply:
Yes. If you wonder why we can not leverage more directly the `sklearn` API, please see the comment above about the 2-way PD for categorical-numerical.